terraform/context.go

```diff
@@ -355,27 +363,33 @@ func (c *Context) Input(mode InputMode) error {
 						"Error asking for %s: %s", n, err)
 				}
 
-				if value == "" && v.Required() {
-					// Redo if it is required, but abort if we keep getting
-					// blank entries
-					if retry > 2 {
-						return fmt.Errorf("missing required value for %q", n)
-					}
-					retry++
-					continue
-				}
-
 				if value == "" {
-					// No value, just exit the loop. With no value, we just
-					// use whatever is currently set in variables.
-					break
+					if v.Required() {
+						// Redo if it is required, but abort if we keep getting
+						// blank entries
+						if retry > 2 {
+							return fmt.Errorf("missing required value for %q", n)
+						}
+						retry++
+						continue
+					}
```

Comment:
I think there was a statement after the nested if during some iteration. I'll collapse this back down again before merging.